core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/source/mysql/MysqlSource.java

```diff
@@ -0,0 +1,161 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.constants.SchemaConstants;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.dataflow.ISink;
+import edu.uci.ics.texera.api.dataflow.ISourceOperator;
+import edu.uci.ics.texera.api.exception.DataflowException;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.field.DoubleField;
+import edu.uci.ics.texera.api.field.IDField;
+import edu.uci.ics.texera.api.field.IField;
+import edu.uci.ics.texera.api.field.IntegerField;
+import edu.uci.ics.texera.api.field.StringField;
+import edu.uci.ics.texera.api.field.TextField;
+import edu.uci.ics.texera.api.field.DateField;
+import edu.uci.ics.texera.api.schema.Attribute;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.dataflow.twitterfeed.TwitterUtils;
+
+public class MysqlSource implements ISourceOperator{
+	private final MysqlSourcePredicate predicate;
+    private int cursor = CLOSED;
+    private Schema outputSchema;
+    private Connection connection;
+    private Statement statement;
+    private PreparedStatement prepStatement;
+    private ResultSet rs;
+    
+    public MysqlSource(MysqlSourcePredicate predicate){
+    	this.predicate = predicate;
+    	//outputSchema = new Schema(inputSchema.getAttributes().stream()
+          //      .filter(attr -> !attr.getName().equalsIgnoreCase(SchemaConstants._ID))
+            //    .filter(attr -> !attr.getName().equalsIgnoreCase(SchemaConstants.PAYLOAD))
+             //   .filter(attr -> !attr.getType().equals(AttributeType.LIST)).toArray(Attribute[]::new));
+    	/*this.outputSchema = new Schema(
+                SchemaConstants._ID_ATTRIBUTE, 
+                new Attribute(this.predicate.getResultAttribute(), AttributeType.STRING));
+    */
+    	this.outputSchema = new Schema.Builder().add(SchemaConstants._ID_ATTRIBUTE)
+                .add(TwitterUtils.TwitterSchema.TWITTER_SCHEMA).build();
+    }
+    
+    @Override
+    public void open() throws TexeraException {
+        if (cursor == OPENED) {
+            return;
+        }
+        // JDBC connection
+        try {
+            Class.forName("com.mysql.jdbc.Driver").newInstance();
+            String url = "jdbc:mysql://" + predicate.getHost() + ":" + predicate.getPort() + "/"
+                    + predicate.getDatabase() + "?autoReconnect=true&useSSL=true";
+            this.connection = DriverManager.getConnection(url, predicate.getUsername(), predicate.getPassword());
+            
+            statement = connection.createStatement();
+            rs = statement.executeQuery(generateSqlQuery(predicate));
```

Comment:
the query should not be executed inside `open()`. It should be executed the first time `getNextTuple` is called. @avinash0161 please check this